docs/data/date-pickers/validation/validation.md

```diff
@@ -0,0 +1,191 @@
+---
+title: Date and Time pickers - Validation
+---
+
+# Date and Time pickers - Validation
+
+<p class="description">Add custom validation to user inputs.</p>
+
+All the date and time pickers have an API for adding validation constraints.
+By default they provide visual feedback if the component value doesn't meet the validation criteria.
+
+:::info
+The validation props are showcased for each type of picker component (`TimePicker`, `DatePicker`, `DateRangePicker`, etc).
+But the same props are available on all component variants.
+
+For example—the validation props available on the `DatePicker` component are also available on:
+
+- `DesktopDatePicker`
+- `MobileDatePicker`
+- `StaticDatePicker`
+
+:::
+
+## Invalid values feedback
+
+On the input—it enables its error state.
+
+{{"demo": "ValidationBehaviorInput.js", "defaultCodeOpen": false}}
+
+On the calendar and clock views—the invalid values are displayed as disabled to prevent their selection.
+
+{{"demo": "ValidationBehaviorView.js", "defaultCodeOpen": false}}
+
+## Date validation
+
+All the props described below are available on all the components supporting date edition.
+
+### Disable past or future dates
+
+The `disablePast` prop prevents the selection of all values before today.
+
+- On the `day` view—all the days before today won't be selectable.
+- On the `month` and `year` views—all the values ending before today won't be selectable.
```

Comment:
PR #6226 is adding the support of time for `disablePast` and `disableFuture`. So the behavior is becoming more generic.

If accepted, this part should be a section (instead of a subsection)

Reply:
Indeed, I will move the section once it is merged